x/account/types/genesis.go

```diff
@@ -22,6 +24,25 @@ func (gs GenesisState) Validate() error {
 	// this line is used by starport scaffolding # ibc/genesistype/validate
 
 	// this line is used by starport scaffolding # genesis/types/validate
+	// Check for duplicated index in coordinatorByAddress
+	coordinatorByAddressIndexMap := make(map[string]struct{})
+
+	for _, elem := range gs.CoordinatorByAddressList {
```

Comment:
This function is used to verify if a genesis file is valid and can be used for a chain. The check is static and depends only on the file itself.
By default, we only add a check that verifies that no duplicated index keys exist in the provided lists.

Additional steps we can do for genesis validation based on our structure is:
- all IDs retrieved from the `CoordinatorByAddress` must exist in `Coordinator`
- All addresses retrieved from the `Coordinator` must exist in `CoordinatorByAddress`

We we can simply reuse the maps in this file and add those verifications